src/MongoDB.Bson/ObjectModel/ObjectId.cs

```diff
@@ -36,7 +36,7 @@ public struct ObjectId : IComparable<ObjectId>, IEquatable<ObjectId>, IConvertib
         private static int __staticIncrement = (new Random()).Next();
 
         // private fields
-        private readonly int _a;
+        private readonly uint _a;
```

Comment:
this is the main change in this PR - replacing the type of `_a` (`timestamp`) from `int` to `uint`

Reply:
Let's not make this change. It's too intrusive.

The type can be left as `int`.

The only change that needs to be made is that `timestamp` is *interpreted* as `uint`... it doesn't actually have to *be* `uint` (which would be a breaking change).